common/djangoapps/student/models.py

```diff
@@ -2750,3 +2754,21 @@ class Meta(object):
         db_table = "auth_accountrecovery"
 
     objects = AccountRecoveryManager()
+
+    @classmethod
+    def retire_recovery_email(cls, user_id):
+        """
+        Retire user's recovery/secondary email as part of GDPR Phase I
+        Returns 'True'
+
+        If an AccountRecovery record is found for this user it will be deleted,
+        if it is not found it is assumed this table has no PII for the given user.
+
+        :param user_id: int
+        :return: bool
+        """
+        try:
+            cls.objects.filter(user_id=user_id).delete()
+            return True
+        except cls.DoesNotExist:
+            return True
```

Comment:
There should only ever be one `AccountRecovery` row for a user. So I would think that you would want to use `get` instead of `filter` here. `filter` won't ever raise `DoesNotExist` error that you are catching, it will just no-op the delete if there are no rows in the queryset.

Tiny nit: You could just `pass` in the except clause and move the return down below to keep just one return branch.

Reply:
Thanks, good points! I'll make those changes